drivers/SmartThings/zigbee-motion-sensor/src/aqara/init.lua

```diff
@@ -0,0 +1,154 @@
+local FREQUENCY_ATTRIBUTE_ID = 0x0102
+local SENSITIVITY_ATTRIBUTE_ID = 0x010C
+local FREQUENCY_DEFAULT_VALUE = 120
+local UNOCCUPIED_TIMER = "unoccupiedTimer"
+local FREQUENCY_PREF = "frequencyPref"
+local SENSITIVITY_HIGH = 3
+local SENSITIVITY_MEDIUM = 2
+local SENSITIVITY_LOW = 1
+local OCCUPANCY_OCCUPIED = 1
+
+local FINGERPRINTS = {
+  { mfr = "LUMI", model = "lumi.motion.agl02" },
+  { mfr = "LUMI", model = "lumi.motion.agl04" }
+}
+
+local is_aqara_products = function(opts, driver, device)
+  for _, fingerprint in ipairs(FINGERPRINTS) do
+    if device:get_manufacturer() == fingerprint.mfr and device:get_model() == fingerprint.model then
+      return true
+    end
+  end
+  return false
+end
+
+local write_motion_pref_attribute = function(device, cluster, attr, value)
+  device:send(cluster_base.write_manufacturer_specific_attribute(device, cluster, attr, MFG_CODE,
+    data_types.Uint8, value))
+end
+
+local function added_handler(self, device)
+  device:emit_event(capabilities.motionSensor.motion.inactive())
+  device:emit_event(sensitivityAdjustment.sensitivityAdjustment.Medium())
+  device:emit_event(detectionFrequency.detectionFrequency(FREQUENCY_DEFAULT_VALUE))
+  device:emit_event(capabilities.battery.battery(100))
+end
+
+local do_configure = function(self, device)
+  device:configure()
+
+  device:send(device_management.build_bind_request(device, OccupancySensing.ID, self.environment_info.hub_zigbee_eui))
+  device:send(OccupancySensing.attributes.Occupancy:configure_reporting(device, 30, 3600))
+
+  device:send(device_management.build_bind_request(device, PowerConfiguration.ID, self.environment_info.hub_zigbee_eui))
+  device:send(PowerConfiguration.attributes.BatteryVoltage:configure_reporting(device, 30, 3600, 1))
+
+  device:send(cluster_base.write_manufacturer_specific_attribute(device, PRIVATE_CLUSTER_ID, PRIVATE_ATTRIBUTE_ID,
+    MFG_CODE, data_types.Uint8, 1))
+end
+
+local function motion_active_handler(driver, device, value, zb_rx)
+  device:emit_event(capabilities.motionSensor.motion.active())
+
+  local unoccupied_timer = device:get_field(UNOCCUPIED_TIMER)
+  if unoccupied_timer then
+    device.thread:cancel_timer(unoccupied_timer)
+    device:set_field(UNOCCUPIED_TIMER, nil)
+  end
+  local detect_duration = device:get_field(FREQUENCY_PREF) or FREQUENCY_DEFAULT_VALUE
+  local inactive_state = function()
+    device:emit_event(capabilities.motionSensor.motion.inactive())
+  end
+  unoccupied_timer = device.thread:call_with_delay(detect_duration, inactive_state)
+  device:set_field(UNOCCUPIED_TIMER, unoccupied_timer)
+end
+
+local function attr_handler(driver, device, value, zb_rx)
+  -- The low 16 bits for Illuminance
+  -- The high 16 bits for Motion Detection
+  if value.value > 65536 then
+    -- active
+    local lux = value.value - 65536
+    device:emit_event(capabilities.illuminanceMeasurement.illuminance(lux))
+
+    motion_active_handler(driver, device, value, zb_rx)
```

Comment:
Is this necessary? there is separate handler(occupancy_attr_handler) for motion detection. If attr_handler/occupancy_attr_handler called separately, motion_active_handler() will be called twice.

Reply:
1) Motion Sensor T1 sends the detected value  and the brightness value when motion is detected. This uses the Private cluster attribute 0x0112.

2) High Precision Motion Sensor sends the detected value only when motion is detected. This uses the OccupancySensing cluster attribute.

3) Therefore, each device is mutually exclusive and the attribute handler is not executed twice.

Reply:
okay, got it.